Brokerages/InteractiveBrokers/InteractiveBrokersBrokerage.cs

```diff
@@ -435,15 +435,26 @@ public override List<CashAmount> GetCashBalance()
 
             if (!IsConnected)
             {
-                Log.Trace("InteractiveBrokersBrokerage.GetCashBalance(): not connected, connecting now");
-                Connect();
+                if (IsWithinScheduledServerResetTimes())
+                {
+                    // Occasionally the disconnection due to the IB reset period might last
+                    // much longer than expected during weekends (even up to the cash sync time).
+                    // In this case we do not try to reconnect (since this would fail anyway)
+                    // but we return the existing balances instead.
+                    Log.Trace("InteractiveBrokersBrokerage.GetCashBalance(): not connected within reset times, returning existing balances");
```

Comment:
`IsWithinScheduledServerResetTimes()` will return true on `Saturdays` and from 11:00pm->1:30am, right?
I'm not sure how will this new `if` statement, cover for this case `...much longer than expected during weekends (even up to the cash sync time)...`? from what I understand it will go through `Connect();` as before

Reply:
I was considering a disconnection from e.g. Friday night at 11 pm to Saturday morning at 7:45 am as `much longer than expected`.
At 7:45, with current master branch code, we are expecting to be able to connect successfully (and it seems to not be always the case).
With this change, we are able to still be disconnected until Monday at 7:45 without killing the algorithm and hopefully, we'll be able to connect then 😃